devito/dimension.py

```diff
@@ -372,14 +372,14 @@ def _arg_defaults(self, **kwargs):
         A :class:`ConditionalDimension` provides no arguments, so this
         method returns an empty dict.
         """
-        return {}
+        return {self.size_name: kwargs.get('size')}
 
     def _arg_values(self, *args, **kwargs):
         """
         A :class:`ConditionalDimension` provides no arguments, so there are
         no argument values to be derived.
         """
-        return {}
+        return kwargs
```

Comment:
I have no idea why you have to do this

Reply:
I think a second pair of eyes would be useful here @tjb900

Reply:
Brute force making work left over to clean up.